backend/Board.py

```diff
@@ -55,37 +30,85 @@ def create_initial_board(self, size):
+            pos_map[(ini, thi)] = [(ini, sec), (sec, thi)]
+
+            pos_map[(sec, ini)] = [(ini, ini), (thi, ini)]
+            pos_map[(sec, thi)] = [(ini, thi), (thi, thi)]
+
+            pos_map[(thi, ini)] = [(sec, ini), (thi, sec)]
+            pos_map[(thi, sec)] = [(thi, ini), (thi, thi)]
+            pos_map[(thi, thi)] = [(thi, sec), (sec, thi)]
+        
+        mid = self.board_info["row_size"]
+        
+        # "positive" interconnections for squares (a4-b4-c4, e.g.)
+        for inter in range(self.board_info["row_size"]-1):
+            pos_map[(mid, inter)].append((mid, inter+1))
+            pos_map[(mid, inter+1)].append((mid, inter))
+
+            pos_map[(inter, mid)].append((inter+1, mid))
+            pos_map[(inter+1, mid)].append((inter, mid))
+        
+        # "negative" interconnections for squares (g4-f4-e4, e.g.)
+        for inter in range(self.board_info["row_size"]-1, 0, -1):
+            fin = mid*2 - inter
+
+            pos_map[(mid, fin)].append((mid, fin+1))
+            pos_map[(mid, fin+1)].append((mid, fin))
+
+            pos_map[(fin, mid)].append((fin+1, mid))
+            pos_map[(fin+1, mid)].append((fin, mid))
+
+        # diagonals
+        if self.board_info["diagonals"]:
+            for inter in range(self.board_info["row_size"]-1):
+                fin = mid*2 - inter
+
+                pos_map[(inter, inter)].append((inter+1, inter+1))
+                pos_map[(inter+1, inter+1)].append((inter, inter))
+
+                pos_map[(fin, fin)].append((fin-1, fin-1))
+                pos_map[(fin-1, fin-1)].append((fin, fin))
+
+                pos_map[(inter, fin)].append((inter+1, fin-1))
+                pos_map[(inter+1, fin-1)].append((inter, fin))
+
+                pos_map[(fin, inter)].append((fin-1, inter+1))
+                pos_map[(fin-1, inter+1)].append((fin, inter))
+
+        return pos_map
     
     def init_valid_moves(self, row):
-        middle = (self.board_size - 1) // 2
+        middle = (self.board_info["size"] - 1) // 2
         if row == middle:
             return self._get_middle_row_moves(middle)
         else:
             return self._get_non_middle_row_moves(row, middle)
 
     def _get_middle_row_moves(self, middle):
         row_moves = []
-        for cell in range(self.board_size):
+        for cell in range(self.board_info["size"]):
             if cell != middle:
                 row_moves.append(cell)
         return row_moves
 
-    # 3 needs to be changed to support any board size
+    '''hmm, so the "magic number" 3 here represents the number valid spots in a row. 
+    that actually doesn't change depending on the board, except in the middle on 6, 
+    but the middle works fine and is in a separate function
+    
+    we can change this if you guys are concerned about it, but it's the same no matter the men's morris game. even 3 works this way'''
     def _get_non_middle_row_moves(self, row, middle):
```

Comment:
Okay, it won't let me comment directly on the line I want to, but this 3 is a magic number. However, it's not one that changes. For all of the variants of Nine Men's Morris on Wikipedia, there are 3 pieces in each row and column of each square. (Where the rows and columns are is a little different -- that changes depending on the board size.) We can assign this a static variable or something at the top, if you want, but it's not used anywhere else and it doesn't change.